newdle/client/src/components/ParticipantTablev2.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import moment from 'moment';
+import {useSelector} from 'react-redux';
+import {Icon, Table} from 'semantic-ui-react';
+import {serializeDate, toMoment} from '../util/date';
+import {
+  getNewdleDuration,
+  getNewdleParticipants,
+  getNewdleTimezone,
+  getNewdleTimeslots,
+} from '../selectors';
+import UserAvatar from './UserAvatar';
+import styles from './creation/userSearch/UserSearch.module.scss';
+import PropTypes from 'prop-types';
+
+function formatMeetingTime(startTime, duration) {
+  const endTime = moment(startTime)
+    .add(duration, 'm')
+    .format('HH:mm');
+  return `${serializeDate(startTime, 'HH:mm')} - ${endTime}`;
+}
+
+function DateCell({timeslot}) {
+  const newdleTimezone = useSelector(getNewdleTimezone);
+  const duration = useSelector(getNewdleDuration);
+
+  const startTime = toMoment(timeslot, 'YYYY-MM-DDTHH:mm');
+  return (
+    <Table.HeaderCell textAlign="center">
+      <div>
+        <div>{startTime.format('D MMM')}</div>
+        <div>{formatMeetingTime(startTime, duration)}</div>
+        <div>({newdleTimezone})</div>
+      </div>
+    </Table.HeaderCell>
+  );
+}
+
+DateCell.propTypes = {
+  timeslot: PropTypes.string,
+};
+
+function ParticipantRow({participant}) {
+  const timeslots = useSelector(getNewdleTimeslots);
+  let participantName = participant.name;
+  let participantEmail = participant.email;
+  // eslint-disable-next-line react/prop-types
+  let matches = participantName.match(/\b(\w)/g);
+  let initials = matches.join('');
+
+  return (
+    <Table.Row textAlign="center">
+      <Table.Cell>
+        <div style={{display: 'flex'}}>
+          <UserAvatar
+            user={{email: participantEmail, name: participantName, initials}}
+            size={25}
+            className={styles['participant-avatar']}
+          />
+          {participantName}
+        </div>
+      </Table.Cell>
+      {timeslots.map((timeslot, index) => {
+        let status = null;
+        if (participant.answers[timeslot] === 'available') {
+          status = 'available';
+        } else if (participant.answers[timeslot] === 'unavailable') {
+          status = 'unavailable';
+        } else if (participant.answers[timeslot] === 'ifneedbe') {
+          status = 'ifneedbe';
+        }
+        return (
+          <Table.Cell key={`${status}-${index}`} textAlign="center">
```

Comment:
using the index in the key is usually not a good idea, what about `key={timeslot}`?